### PR TITLE
[JENKINS-34877] Fix PCT against 2.x and migrate to 2.9 parent pom

### PR DESCRIPTION
**Pull request by armfergom**
[JENKINS-34877](https://issues.jenkins-ci.org/browse/JENKINS-34877)

- [x] Migrate to 2.9 parent pom
- [x] Fixes PCT against 2.x baseline
- [x] Update `.gitignore` with Eclipse files.
- [x] Add basic test suite

@reviewbybees

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/plaincredentials/BaseTest.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    private static final String CRED_ID = "Custom-ID";
+
+    @Rule
+    public JenkinsRule r = new JenkinsRule();
+    
+    private CredentialsStore store;
+    
+    @Before
+    public void setup(){
+        store = CredentialsProvider.lookupStores(r.jenkins).iterator().next();;
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/plaincredentials/BaseTest.java]: doubled `;;`

File: src/test/java/org/jenkinsci/plugins/plaincredentials/BaseTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    }
+    
+    /**
+     * Creates, updates and deletes credentials and perform different assertions
+     * 
+     * @param credential the credential to create
+     * @param updatedCredential the credential that will replace the first one during update
+     * @throws IOException
+     */
+    private <T extends BaseStandardCredentials> void testCreateUpdateDetele(T credential, T updatedCredential) throws IOException {
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/plaincredentials/BaseTest.java]: typo

File: src/test/java/org/jenkinsci/plugins/plaincredentials/BaseTest.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    public void secretFileBaseTest() throws IOException, URISyntaxException {
+        DiskFileItem fileItem = createEmptyFileItem();
+        
+        FileCredentialsImpl credential = new FileCredentialsImpl(CredentialsScope.GLOBAL, CRED_ID, "Test Secret file", fileItem, "keys.txt", Base64.encode(fileItem.get()).toString());
+        FileCredentialsImpl updatedCredential = new FileCredentialsImpl(credential.getScope(), UPDATED_CRED_ID, credential.getDescription(), fileItem, credential.getFileName(), credential.getData());
+        testCreateUpdateDetele(credential, updatedCredential);
+    }
+    
+    /**
+     * Creates, updates and deletes credentials and perform different assertions
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/plaincredentials/BaseTest.java]: Not really sure what the point of these assertions is. Are you actually testing any code from this plugin? Or could this all be moved into `credentials-plugin`, for example using `UsernamePasswordCredentialsImpl`?

**Comment by jglick** [comment]
:bee:

**Comment by armfergom** [comment]
I wanted to write tests to ensure that these two types of credentials can be created with no issue. The code from this plugin that is being tested is basically the constructors. 
But it is true that this mainly tests code from the credentials plugin, which already has similar tests in place. Happy to remove it if it needs be.

**Comment by jglick** [comment]
Will leave it to you to determine whether the newly added tests are really extending coverage of anything interesting. I am _guessing_ they are not, since the tricky part specific to this plugin is the GUI for creating file credentials (which this does not simulate); but the tests seem harmless so there is no particular need to delete them unless they become a maintenance burden. re:bee:

**Pull request merged**